registry/handlers/images.go

```diff
@@ -83,6 +91,64 @@ func (imh *imageManifestHandler) GetImageManifest(w http.ResponseWriter, r *http
 		return
 	}
 
+	supportsSchema2 := false
```

Comment:
Shouldn't this fallback only happen when fetching by tag? Am I understanding this correctly?